### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by mloehnig** [3d0c/gmf#157]
Hello, 
I just started to use this package and I stumbled across some interesting behavior, I noticed that whenever a `syscall.EAGAIN` is received no error is returned and potentially a nil value is also returned. [Example 1](https://github.com/3d0c/gmf/blob/be727bc5b56cc2ab61e39d2a16c0e5ed0d00b82e/codecCtx_go112.go#L591) [Example 2](https://github.com/3d0c/gmf/blob/be727bc5b56cc2ab61e39d2a16c0e5ed0d00b82e/frame_go112.go#L88)

Typically in Go you would propagate an error like this to allow the application layer to be able to change behavior. 

Is there a particular reason for not propagating the error?

**Comment by 3d0c** [comment]
Here is a comment from `decode.h` of `libavcodec`
```
/**
 * Called by decoders to get the next packet for decoding.
 *
 * @param pkt An empty packet to be filled with data.
 * @return 0 if a new reference has been successfully written to pkt
 *         AVERROR(EAGAIN) if no data is currently available
 *         AVERROR_EOF if and end of stream has been reached, so no more data
 *                     will be available
 */
```
This is why.

**Comment by mloehnig** [comment]
Yes I understand the comment. The issue I am having is these two errors are being masked/not returned or propagated. If the error EAGAIN, error i am receiving, is received my application code should retry again but since I never received the error I have no way of making a smart application level decision. I would happily make a PR to allow for these values to be propagated if you agree they should be.

**Comment by 3d0c** [comment]
Feel free to contribute, would be great to have another one Decode function, let's name it Decode2 for backward compatibility
